### ISSUE TITLE
Incomplete password redaction

### ISSUE THREAD

**Issue by whorfin** [navidrome/navidrome#1103]
### Description
Thanks much for the Password redaction feature, this is a nice improvement for security.
Unfortunately it has a bug.
If a password contains spaces, only the first "word" is redacted, at least in the scenario below.

Here I'm using iSub on iOS, which very much likes to fetch lyrics all the time.
This causes the following log messages where the user was "user_name" and password was "first and other words":

time="2021-05-07T17:59:24-07:00" level=error msg="HTTP: GET http://myserver.notreally.com/rest/getLyrics.view?v=1.2.0&c=iSub&u=user_name&p=[REDACTED]%20and%20other%20words&title=The%20Drift&artist=Erik%20W%C3%B8llo" elapsedTime="548.932µs" httpStatus=510 remoteAddr=192.168.0.100 requestId=server/FgdfkZTzkF-000106 responseSize=63

### Expected Behaviour
I think we all would greatly prefer that "[REDACTED]" replaced the _entire_ password

### Steps to reproduce
Add a user with spaces in their password
Access navidrome using a client which looks for lyrics, or some other method which shows a log msg with the un-fully-redacted password

### Platform information

 - Navidrome version: 0.42.1-SNAPSHOT (3e8d3e78)
 - Browser and version:  iSub and others
 - Operating System: built from source on illumos

**Comment by whorfin** [comment]
At a guess, the issue is the definition of a password in `log/log.go` as a sequence of words; this allows escaped specials, such as space, to not match as eg the percent in `%20`  isn't a word character.
At the risk of attempting to be helpful, perhaps 
```
         "([^\\w]p=)[\\w%]+",
```
instead of 
```
         "([^\\w]p=)[\\w]+",
```
But of course you Should Not Trust Me; the redaction string needs to be validated against valid input,, with particular care taken for special characters.  Perhaps adopting a more "greedy" mechanism  would be better -- if I read the redactrus docs correctly,  something like this might work?
```
```
         "([^\\w]p=)[^&]+(&)",
```
```

**Comment by deluan** [comment]
Good catch. Please check if the latest fix the issue. If not, feel free to reopen the issue
